### ISSUE TITLE
`conda activate` sometimes fails

### ISSUE THREAD

**Issue by DavidMertz** [conda/conda#5906]
There's something oddly sequence dependent on when `conda activate` works or fails.  This is an environment that I've switched back and forth from using regular `conda [de]activate` multiple times.  Not sure exactly what series of other conda commands have happened since it last worked.

```bash
504-~ % conda activate pandas1
Traceback (most recent call last):
  File "//anaconda/lib/python3.6/site-packages/conda/cli/main.py", line 176, in main
    raise CommandNotFoundError(argv1)
conda.exceptions.CommandNotFoundError: 'activate is not a conda command.
Did you mean 'source activate'?


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "//anaconda/bin/conda", line 6, in <module>
    sys.exit(conda.cli.main())
  File "//anaconda/lib/python3.6/site-packages/conda/cli/main.py", line 178, in main
    from ..exceptions import handle_exception
ImportError: cannot import name 'handle_exception'
505=~ % conda activate pandas1
[... identical behavior ...]
506-~ % source activate pandas1
(pandas1) 507-~ % conda deactivate
508-~ % conda activate pandas1
(pandas1) 509-~ % conda deactivate 
```

**Comment by DavidMertz** [comment]
To be complete:

```bash
516-~ % conda info

     active environment : None
            shell level : 0
       user config file : /Users/dmertz/.condarc
 populated config files : /Users/dmertz/.condarc
          conda version : 4.4.0rc1
    conda-build version : 2.1.5
         python version : 3.6.1.final.0
       base environment : //anaconda  (writable)
           channel URLs : https://conda.anaconda.org/conda-canary/osx-64
                          https://conda.anaconda.org/conda-canary/noarch
                          https://conda.anaconda.org/anaconda-fusion/osx-64
                          https://conda.anaconda.org/anaconda-fusion/noarch
                          https://conda.anaconda.org/anaconda-mosaic/osx-64
                          https://conda.anaconda.org/anaconda-mosaic/noarch
                          https://conda.anaconda.org/R/osx-64
                          https://conda.anaconda.org/R/noarch
                          https://repo.continuum.io/pkgs/free/osx-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/osx-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/osx-64
                          https://repo.continuum.io/pkgs/pro/noarch
          package cache : //anaconda/pkgs
                          /Users/dmertz/.conda/pkgs
       envs directories : //anaconda/envs
                          /Users/dmertz/.conda/envs
               platform : osx-64
             user-agent : conda/4.4.0rc1 requests/2.14.2 CPython/3.6.1 Darwin/16.7.0 OSX/10.12.6
                UID:GID : 502:20
             netrc file : None
           offline mode : False
```

**Comment by mingwandroid** [comment]
Hi @DavidMertz, somehow (I've seen this before), your installation thinks it's installed at `//` and that causes a lot of things to go wrong, it could be the root of your problems actually.

**Comment by DavidMertz** [comment]
Doing the system wide installation is an important use case though, isn't it?

I could reinstall for just one user, of course... but surely many non-Anaconda employees will choose that "all users" installation option.

**Comment by mingwandroid** [comment]
It's not the system wide aspect I was alluding to here, it's the completely bogus double '/' at the front.

**Comment by mingwandroid** [comment]
Hopefully before the next release we'll be able to figure this out. Did you install via the `.pkg` graphical installer?

**Comment by DavidMertz** [comment]
Unfortunately, I've done several reinstalls of Anaconda over time.  I **think** the most recent one was the command-line version, but I might be misremembering.  Is there something I can look at to figure out which installer was used most recently?

**Comment by DavidMertz** [comment]
I've noticed the bogus `//` since I started using Anaconda. It's definitely been there on every system-wide installation I've seen.  It seemed wrong, but it also seemed like it shouldn't matter, given:

```bash
501-~ % ls //
Applications              anaconda                  net
[...]
Users                     home                      usr
Volumes                   installer.failurerequests var
```

Or:

```python
In [1]: import glob
In [2]: glob.glob('//*')
Out[2]:
['//anaconda',
 '//Applications',
 '//bin',
[...]
 '//Users',
 '//usr',
 '//var',
 '//Volumes']
```

**Comment by mingwandroid** [comment]
In the conda-meta folder there are a load of .json files. They should say 'installed by' ...

**Comment by DavidMertz** [comment]
```bash
520-~ % for f in $(find /anaconda -name 'conda-meta'); do grep 'installed_by' $f/*.json; done | head -5
/anaconda/conda-meta/_license-1.1-py36_1.json:  "installed_by": "Anaconda3-4.3.0-MacOSX-x86_64",
/anaconda/conda-meta/appnope-0.1.0-py36_0.json:  "installed_by": "Anaconda3-4.3.0-MacOSX-x86_64",
/anaconda/conda-meta/appscript-1.0.1-py36_0.json:  "installed_by": "Anaconda3-4.3.0-MacOSX-x86_64",
/anaconda/conda-meta/astroid-1.4.9-py36_0.json:  "installed_by": "Anaconda3-4.3.0-MacOSX-x86_64",
/anaconda/conda-meta/backports-1.0-py36_0.json:  "installed_by": "Anaconda3-4.3.0-MacOSX-x86_64",
521-~ % for f in $(find /anaconda -name 'conda-meta'); do grep 'installed_by' $f/*.json; done | cut -d: -f3 | sort -u
 "Anaconda3-4.3.0-MacOSX-x86_64",
```

**Comment by mingwandroid** [comment]
Ah, hmm, looks like the exact installer is not recorded.

I could show loads of instances of code that could mishandle `//` (heck I've undoubtable written some myself) and have helped out on GitHub issues where this was determined to be the likely cause.

Can you show the output from `conda config --show-source`

**Comment by DavidMertz** [comment]
```bash
522-~ % conda config --show-source

==> /Users/dmertz/.condarc <==
concurrent: True
ssl_verify: True
channels:
  - conda-canary
  - anaconda-fusion
  - anaconda-mosaic
  - R
  - defaults
anaconda_upload: True
```

**Comment by mingwandroid** [comment]
Is there anything starting with `//` in your `env`? I'm running out of ideas here. @kalefranz, can you suggest anything?

**Comment by DavidMertz** [comment]
Why yes there is:

```bash
522-~ % env | grep '//'
PATH=//anaconda/bin:/usr/local/bin:/sw/bin:/usr/local/sbin:/Users/dmertz/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/git/bin:/usr/local/MacGPG2/bin:/Library/TeX/texbin:/Users/dmertz:/Users/dmertz/Projects/training-tools/
```

**Comment by DavidMertz** [comment]
Given that the problem is intermittent not obvious to replicate, if I modified `.bash_profile` that wouldn't make it immediately clear that such was the solution.  Although if I saw the problem even after making the change, it would show the contrary.

**Comment by mingwandroid** [comment]
No, but I'm going to make a bug for myself to make sure we do not emit these non-standard entries to `PATH`.

**Comment by DavidMertz** [comment]
OK... I think I can rule out the `//` as being the problem.  I modified `.bash_profile` manually to only have one slash.  I launched a new terminal (not for this issue, for other reasons).  With no double-slash anywhere in my environment I still see the problem.  Interestingly, the traceback still acts as if there is a double slash in there though.

```bash
501-~ % conda activate module-machine-learning
Traceback (most recent call last):
  File "//anaconda/lib/python3.6/site-packages/conda/cli/main.py", line 176, in main
    raise CommandNotFoundError(argv1)
conda.exceptions.CommandNotFoundError: 'activate is not a conda command.
Did you mean 'source activate'?


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/anaconda/bin/conda", line 6, in <module>
    sys.exit(conda.cli.main())
  File "//anaconda/lib/python3.6/site-packages/conda/cli/main.py", line 178, in main
    from ..exceptions import handle_exception
ImportError: cannot import name 'handle_exception'
502-~ % echo $PATH
/anaconda/bin:/usr/local/bin:/sw/bin:/usr/local/sbin:/Users/dmertz/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/usr/local/git/bin:/usr/local/MacGPG2/bin:/Library/TeX/texbin:/Users/dmertz:/Users/dmertz/Projects/training-tools/
503-~ % cat ~/.bash_profile
. $HOME/.bashrc

# added by Anaconda3 4.3.0 installer
export PATH="/anaconda/bin:$PATH"

test -e "${HOME}/.iterm2_shell_integration.bash" && source "${HOME}/.iterm2_shell_integration.bash"
```

**Comment by DavidMertz** [comment]
And just to mention, as before if I `source activate` anything, afterwards `conda activate` starts working again.

**Comment by gschivley** [comment]
I've had a similar issue and no `//` in sight

```
Gregs-MacBook-Pro:~ Home$ conda activate ccs-timing
Traceback (most recent call last):
  File "/Users/Home/anaconda/bin/conda", line 6, in <module>
    sys.exit(conda.cli.main())
  File "/Users/Home/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 178, in main
    from ..exceptions import handle_exception
ImportError: cannot import name handle_exception
Gregs-MacBook-Pro:~ Home$ source activate ccs-timing
(ccs-timing) Gregs-MacBook-Pro:~ Home$ conda deactivate
Gregs-MacBook-Pro:~ Home$ conda activate ccs-timing
(ccs-timing) Gregs-MacBook-Pro:~ Home$ conda info

     active environment : ccs-timing
    active env location : /Users/Home/anaconda/envs/ccs-timing
            shell level : 1
       user config file : /Users/Home/.condarc
 populated config files : /Users/Home/anaconda/.condarc
                          /Users/Home/.condarc
          conda version : 4.4.0rc1
    conda-build version : 1.20.3
         python version : 2.7.13.final.0
       base environment : /Users/Home/anaconda  (writable)
           channel URLs : https://conda.anaconda.org/conda-canary/osx-64
                          https://conda.anaconda.org/conda-canary/noarch
                          https://repo.continuum.io/pkgs/free/osx-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/osx-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/osx-64
                          https://repo.continuum.io/pkgs/pro/noarch
          package cache : /Users/Home/anaconda/pkgs
                          /Users/Home/.conda/pkgs
       envs directories : /Users/Home/anaconda/envs
                          /Users/Home/.conda/envs
               platform : osx-64
             user-agent : conda/4.4.0rc1 requests/2.18.4 CPython/2.7.13 Darwin/17.2.0 OSX/10.13.1
                UID:GID : 501:20
             netrc file : None
           offline mode : False
 
```

**Comment by kalefranz** [comment]
Has there been progress here with the release of conda 4.4.0rc2?  We did some work to clean up some of these error messages.  I hoping we got it.  Could use confirmation though.

**Comment by gschivley** [comment]
`conda update conda` isn't finding rc2.

```
~ Home$ conda update conda
Solving environment: done

# All requested packages already installed.
```

Is there something else I need to do?

**Comment by kalefranz** [comment]
conda config --add channels conda-canary
    conda update conda

**Comment by gschivley** [comment]
Already had conda-canary as a channel (see above and below). Tried again, still no luck. Any other info I can provide to help?

```
Gregs-MacBook-Pro:~ Home$ conda config --add channels conda-canary
Warning: 'conda-canary' already in 'channels' list, moving to the top
Gregs-MacBook-Pro:~ Home$ conda info

     active environment : None
       user config file : /Users/Home/.condarc
 populated config files : /Users/Home/anaconda/.condarc
                          /Users/Home/.condarc
          conda version : 4.4.0rc1
    conda-build version : 1.20.3
         python version : 2.7.13.final.0
       base environment : /Users/Home/anaconda  (writable)
           channel URLs : https://conda.anaconda.org/conda-canary/osx-64
                          https://conda.anaconda.org/conda-canary/noarch
                          https://repo.continuum.io/pkgs/free/osx-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/r/osx-64
                          https://repo.continuum.io/pkgs/r/noarch
                          https://repo.continuum.io/pkgs/pro/osx-64
                          https://repo.continuum.io/pkgs/pro/noarch
          package cache : /Users/Home/anaconda/pkgs
                          /Users/Home/.conda/pkgs
       envs directories : /Users/Home/anaconda/envs
                          /Users/Home/.conda/envs
               platform : osx-64
             user-agent : conda/4.4.0rc1 requests/2.18.4 CPython/2.7.13 Darwin/17.2.0 OSX/10.13.1
                UID:GID : 501:20
             netrc file : None
           offline mode : False

Gregs-MacBook-Pro:~ Home$ conda update conda
Solving environment: done

# All requested packages already installed.

Gregs-MacBook-Pro:~ Home$ conda install -c conda-canary conda
Solving environment: done

# All requested packages already installed.
```

**Comment by kalefranz** [comment]
Oh that's interesting.  I understand now.

    conda install -c conda-canary -c https://repo.continuum.io/pkgs/main conda

**Comment by DavidMertz** [comment]
OK... this is weird, I'll put it in a separate issue, but:

```bash
% conda install -c conda-canary -c https://repo.continuum.io/pkgs/main conda
Fetching package metadata ...
............
Solving package specifications: .

Package plan for installation in environment /Users/dmertz/anaconda3:

The following packages will be UPDATED:

    conda:   4.3.30-py36h173c244_0 --> 4.4.0rc2-py36h0a71a83_0 conda-canary
    pycosat: 0.6.2-py36h1486600_0  --> 0.6.3-py36hee92d8f_0

Proceed ([y]/n)?
pycosat-0.6.3- 100% |##############################| Time: 0:00:00   1.55 MB/s
conda-4.4.0rc2 100% |##############################| Time: 0:00:00 906.88 kB/s
```

So far, so good. What environments do I have:

```bash
% conda env list
# conda environments:
#
base                  *  /Users/dmertz/anaconda3
anaconda-project         /Users/dmertz/anaconda3/envs/anaconda-project
my-env                   /Users/dmertz/anaconda3/envs/my-env
py34                     /Users/dmertz/anaconda3/envs/py34
test                     /Users/dmertz/anaconda3/envs/test
```

Something's missing!

```bash
% ls ~/anaconda3/envs/anaconda-project          module-numba              projects-data-io
magic-tools               module-pandas             projects-machine-learning
module-bigdata            module-scipy              py34
module-dask               module-stats              test
module-machine-learning   my-env
module-maintenance        projects-data-analysis
```

**Comment by gschivley** [comment]
I've now installed conda 4.4.0rc3. First try to use `conda activate <env>` gave me an error message asking that I run the command below. After doing so, `conda activate <env>` started working.

```
echo ". /Users/Home/anaconda/etc/profile.d/conda.sh" >> ~/.bash_profile
```

**Comment by kalefranz** [comment]
Can this issue be closed now?  I think we've also resolved with `conda info -e` problem in 4.4.0rc3.
